src/runtime/entity.ts

```diff
@@ -72,7 +72,9 @@ class EntityInternals {
   }
 
   getId(): string {
-    assert(this.isIdentified(), 'getId() called on unidentified entity');
+    if (this.id === undefined) {
```

Comment:
why is `throw new Error` better than an assert, 
and separately why did you not use `this.isIdentified()`

Reply:
undefined narrowing again.  If you check this.id for undefined directly the compiler will  know that it is defined.  If you don't have this the null checking fails as this.id does not match the return type.

Another fix for this would look like:

```
 getId(): string {
    return checkDefined(this.id);
 }
```

This avoids the if-assert block and moves the check and null/defined-ness narrowing into checkDefined()